app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/vm.rb

```diff
@@ -97,6 +102,100 @@ def params_for_create_snapshot
     }
   end
 
+  def params_for_resize
+    {
+      :fields => [
+        {
+          :component         => 'select',
+          :name              => 'pinning',
+          :id                => 'pinning',
+          :label             => _('Pinning'),
+          :initializeOnMount => true,
+          :initialValue      => form_default_values.find_by(:name => 'pin_policy').value,
+          :isRequired        => true,
+          :options           => [
+            {
+              :label => 'None',
+              :value => 'none',
+            },
+            {
+              :label => 'Hard',
+              :value => 'hard',
+            },
+            {
+              :label => 'Soft',
+              :value => 'soft',
+            },
+          ]
+        },
+        {
+          :component         => 'text-field',
+          :name              => 'cores',
+          :id                => 'cores',
+          :label             => _('Cores'),
+          :initializeOnMount => true,
+          :initialValue      => form_default_values.find_by(:name => 'entitled_processors').value,
+          :type              => 'number',
+          :min               => 0.25,
+          :max               => 6.25,
```

Comment:
Is this documented anywhere or did you just see what you could pick in the PowerVS UI?
This is something we'll have to keep updated as PowerVS might allow e.g. 8.25 tomorrow. (Same for memory below).
We might want to consider moving this to `config/settings.yml` so that customers can change it before we can push a new release

Reply:
> in the PowerVS UI it changes everyday

:scream: that's not great, @jaywcarman what do you think about making this a setting vs not having a maximum and having the user hit a failure if they picked a number over whatever max powervs feels like that day?

Reply:
I think we should remove the max and let the request fail if the service doesn't have available capacity (same behavior as our VM provisioning in PowerVS).

As a future enhancement, we would want to use the [PCloudSystemPools API](https://cloud.ibm.com/apidocs/power-cloud#pcloud-systempools-get) to query core and memory capacity on the fly. I'm assuming this is what the IBM Cloud web UI is doing.

Reply:
@agrare @jaywcarman 
Any reason not implement this check now. This seems to work:
`  def params_for_resize
    {
      :fields => [
        {
          :component         => 'select',
          :name              => 'pin_policy',
          :id                => 'pin_policy',
          :label             => _('Pinning'),
          :initializeOnMount => true,
          :initialValue      => form_default_values.find_by(:name => 'pin_policy').value,
          :isRequired        => true,
          :options           => [
            {
              :label => 'None',
              :value => 'none',
            },
            {
              :label => 'Hard',
              :value => 'hard',
            },
            {
              :label => 'Soft',
              :value => 'soft',
            },
          ]
        },
        {
          :component         => 'text-field',
          :name              => 'processors',
          :id                => 'processors',
          :label             => _('Cores'),
          :initializeOnMount => true,
          :initialValue      => form_default_values.find_by(:name => 'entitled_processors').value,
          :type              => 'number',
          :min               => 0.25,
          :max               => 6.25,
          :step              => 0.25,
          :isRequired        => true,
          :validate          => [
            {:type => 'required'},
            {:type => 'min-number-value', :value => 0.25, :message => _("Size must be greater than or equal to .25")},
            {:type => 'max-number-value', :value => max_available.cores, :message => _("Size must be lower than or equal to #{max_available.cores}")}
          ],
        },
        {
          :component         => 'text-field',
          :name              => 'memory',
          :id                => 'memory',
          :label             => _('Memory (GiB)'),
          :initializeOnMount => true,
          :initialValue      => hardware.memory_mb / 1024,
          :type              => 'number',
          :min               => 2,
          :max               => 807,
          :step              => 1,
          :isRequired        => true,
          :validate          => [
            {:type => 'required'},
            {:type => 'min-number-value', :value => 2, :message => _("Size must be greater than or equal to 2")},
            {:type => 'max-number-value', :value => max_available.memory, :message => _("Size must be lower than or equal to #{max_available.memory}")}
          ],

        },
        {
          :component         => 'radio',
          :name              => 'proc_type',
          :id                => 'proc_type',
          :label             => _('Core Type'),
          :initializeOnMount => true,
          :initialValue      => form_default_values.find_by(:name => 'processor_type').value,
          :isRequired        => true,
          :options           => [
            {
              :label => 'Shared uncapped',
              :value => 'shared',
            },
            {
              :label => 'Shared capped',
              :value => 'capped',
            },
            {
              :label => 'Dedicated',
              :value => 'dedicated',
            },
          ]
        },
      ],
    }
  end

  def max_available
    @max_available ||= system_pool[flavor.name.to_sym].max_available
  end

  def system_pool
    @system_pool ||= with_provider_connection(:service => 'PCloudSystemPoolsApi') do |api|
      api.pcloud_systempools_get(cloud_instance_id)
    end
  end`

Reply:
@alizapeikes the issue with running API calls against the provider from the context of DDF `params_for_*` methods is related to zones and network access restrictions.

MIQ has the concept of zones and roles and customers/users can deploy servers or label nodes in such a way that access to network resources can be controlled/limited.

A real world example might be:
1. All outbound internet access is disabled by default
2. Create an "IBM Cloud Providers" Zone
3. Servers in this zone have cloud.ibm.com allowed, event, inventory, and operations roles enabled for these servers
4. API workers do not have outbound internet access

Under this deployment model (which is actually quite common) this API call will be denied.

The options are either
1. Fetch the values required as part of inventory refresh and persist in the database, this can be accessed from API workers without issue
2. Queue a task which runs this API call in the right zone with the ems_operations role and then fetch the task results.

Agree with @jaywcarman I think for now it is simpler to remove the max validation and allow it to fail (assuming the error message is helpful to the user)

Reply:
ok, understood. Will remove the max validation.